rollup/sync_service.go

```diff
@@ -293,147 +293,155 @@ func (s *SyncService) Stop() error {
 func (s *SyncService) VerifierLoop() {
 	log.Info("Starting Verifier Loop", "poll-interval", s.pollInterval, "timestamp-refresh-threshold", s.timestampRefreshThreshold)
 	for {
-		// The verifier polls for ctc transactions.
-		// the ctc transactions are extending the chain.
-		latest, err := s.client.GetLatestTransaction()
-		if err != nil {
-			log.Error("Cannot fetch transaction")
-			continue
+		if err := s.verify(); err != nil {
+			log.Error("Could not verify", "error", err)
 		}
+		time.Sleep(s.pollInterval)
+	}
+}
 
-		if latest == nil {
-			time.Sleep(s.pollInterval)
+func (s *SyncService) verify() error {
+	// The verifier polls for ctc transactions.
+	// the ctc transactions are extending the chain.
+	latest, err := s.client.GetLatestTransaction()
+	if err != nil {
+		return err
+	}
+
+	if latest == nil {
+		return errors.New("latest transaction not found")
+	}
+
+	var start uint64
+	if s.GetLatestIndex() == nil {
+		start = 0
+	} else {
+		start = *s.GetLatestIndex() + 1
+	}
+	end := *latest.GetMeta().Index
+	log.Info("Polling transactions", "start", start, "end", end)
+	for i := start; i <= end; i++ {
+		tx, err := s.client.GetTransaction(i)
+		if err != nil {
+			log.Error("cannot get tx in loop", "error", err)
 			continue
 		}
 
-		var start uint64
-		if s.GetLatestIndex() == nil {
-			start = 0
-		} else {
-			start = *s.GetLatestIndex() + 1
-		}
-		end := *latest.GetMeta().Index
-		log.Info("Polling transactions", "start", start, "end", end)
-		for i := start; i <= end; i++ {
-			tx, err := s.client.GetTransaction(i)
-			if err != nil {
-				log.Error("Cannot get tx in loop", "index", i)
-				continue
-			}
-			log.Debug("Applying transaction", "index", i)
-			err = s.maybeApplyTransaction(tx)
-			if err != nil {
-				log.Error("Cannot apply transaction", "msg", err)
-			}
-			s.SetLatestIndex(&i)
+		log.Debug("Applying transaction", "index", i)
+		err = s.maybeApplyTransaction(tx)
+		if err != nil {
+			log.Error("could not apply transaction", "error", err)
```

Comment:
We want to return in this case so that `s.SetLatestIndex` isn't called, otherwise the transaction will be skipped

Reply:
Got it, FYI is a bug in `master` right now too then https://github.com/ethereum-optimism/go-ethereum/blob/master/rollup/sync_service.go#L326

Reply:
In a similar vein, shouldn't we returning errors instead of `continue`'ing [here](https://github.com/ethereum-optimism/go-ethereum/blob/master/rollup/sync_service.go#L377)? Otherwise won't older transactions get skipped?

Reply:
https://github.com/ethereum-optimism/go-ethereum/pull/289/commits/fc0105374858b618f1eb492f3c6ab0be0cff0f71 fixed here, see commit message for rationale / explanation.